common/test/acceptance/tests/studio/test_studio_components.py

```diff
@@ -65,3 +66,130 @@ def test_add_advanced_problem(self, component):
         add_component(page, 'problem', component, is_advanced_problem=True)
         problem = page.xblocks[1]
         self.assertEqual(problem.name, component)
+
+
+class ComponentTest(ContainerBase):
+    """
+    Test class to add different components.
+    (Not the advanced components)
+    """
+    def setUp(self, is_staff=True):
+        """
+        Create a course with a section, subsection, and unit to which to add the component.
+        """
+        super(ComponentTest, self).setUp(is_staff=is_staff)
+        self.advanced_settings = AdvancedSettingsPage(
+            self.browser,
+            self.course_info['org'],
+            self.course_info['number'],
+            self.course_info['run']
+        )
+
+    def populate_course_fixture(self, course_fixture):
+        course_fixture.add_advanced_settings(
+            {u"advanced_modules": {"value": ["split_test"]}}
+        )
+
+        course_fixture.add_children(
+            XBlockFixtureDesc('chapter', 'Test Section').add_children(
+                XBlockFixtureDesc('sequential', 'Test Subsection').add_children(
+                    XBlockFixtureDesc('vertical', 'Test Unit')
+                )
+            )
+        )
+
+    def asset_components_are_in_order(self, problems, order):
+        """
+        Assert that components appear in the order of their addition
+        """
+        problem_name = []
+        for problem in problems:
+            problem_name.append(problem.name)
+
+        self.assertEquals(problem_name, order)
+
+    def test_add_html_component(self):
+        """
+        Scenario: I can add HTML components
+        Given I am in Studio editing a new unit
+        When I add this type of HTML component:
+            | Component               |
+            | Text                    |
+            | Announcement            |
+            | Zooming Image Tool      |
+            | Raw HTML                |
+        Then I see HTML components in this order:
+            | Component               |
+            | Text                    |
+            | Announcement            |
+            | Zooming Image Tool      |
+            | Raw HTML                |
+        """
+        # Components to be added
+        components = ['Text', 'Announcement', 'Zooming Image Tool', 'Raw HTML']
+        self.go_to_unit_page()
+        page = ContainerPage(self.browser, None)
+        # Add components
+        add_components(page, 'html', components)
+        problems = page.xblocks[1:]
```

Comment:
You could do this in a more straightforward way using list comprehension. Then you wouldn't need the assert_components_are_in_order method. See http://www.diveintopython.net/native_data_types/mapping_lists.html

I don't have devstack running right now so I can't test this out for the exact syntax, but something like:
```
problems = [xblock.name for xblock in page.xblocks[1:]]
self.assertEqual(problems, components)
```